pkg/apis/core/validation/validation.go

```diff
@@ -4294,6 +4302,73 @@ func ValidatePodSpec(spec *core.PodSpec, podMeta *metav1.ObjectMeta, fldPath *fi
 	return allErrs
 }
 
+func validatePodResources(spec *core.PodSpec, podClaimNames sets.Set[string], fldPath *field.Path, opts PodValidationOptions) field.ErrorList {
+	if spec.Resources == nil {
+		return nil
+	}
+
+	allErrs := field.ErrorList{}
+
+	// ValidateResourceRequirements checks if resource names and quantities are
+	// valid, and requests are less than limits.
+	allErrs = append(allErrs, ValidatePodResourceRequirements(spec.Resources, podClaimNames, fldPath, opts)...)
```

Comment:
@kannon92  ValidatePodResourceRequirements -> validatePodResourceName:This is where the resource names are validated. The pod-level resources consistency is checked after that